### PR TITLE
Update pulumi deps.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -3556,7 +3556,7 @@ func Provider() tfbridge.ProviderInfo {
 				ObjectTypeSpec: schema.ObjectTypeSpec{
 					Type: "string",
 				},
-				Enum: []*schema.EnumValueSpec{
```
Comment (t0yv0) [inline provider/resources.go:3559]: I'd like these changes on master please so it's easier to compile pulumi-aws against local pulumi :)

File: sdk/go/aws/apigateway/stage.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ type stageState struct {
 	// The identifier of a client certificate for the stage.
 	ClientCertificateId *string `pulumi:"clientCertificateId"`
 	// The ID of the deployment that the stage points to
-	Deployment *string `pulumi:"deployment"`
+	Deployment interface{} `pulumi:"deployment"`
```
Comment (t0yv0) [inline sdk/go/aws/apigateway/stage.go:159]: This looks a bit concerning, less precise type or is intentional?

Comment (pgavlin) [reply sdk/go/aws/apigateway/stage.go:159]: This is intentional--the original type was incorrect with respect to the actual types of the state inputs.

File: sdk/dotnet/Outputs/ProviderAssumeRole.cs [hunk 3]
```diff
@@ -1,53 +0,0 @@
-// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
-// *** Do not edit by hand unless you're certain you know what you are doing! ***
-
-using System;
-using System.Collections.Generic;
-using System.Collections.Immutable;
-using System.Threading.Tasks;
-using Pulumi.Serialization;
-
-namespace Pulumi.Aws.Outputs
```
Comment (t0yv0) [inline sdk/dotnet/Outputs/ProviderAssumeRole.cs:10]: Interestingly enough this was not referenced anywhere so a safe deletion?

File: sdk/dotnet/S3/Inputs/BucketWebsiteGetArgs.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public sealed class BucketWebsiteGetArgs : Pulumi.ResourceArgs
         /// describing redirect behavior and when redirects are applied.
         /// </summary>
         [Input("routingRules")]
-        public InputUnion<string, ImmutableArray<string>>? RoutingRules { get; set; }
+        public InputUnion<string, ImmutableArray<Input<string>>>? RoutingRules { get; set; }
```
Comment (t0yv0) [inline sdk/dotnet/S3/Inputs/BucketWebsiteGetArgs.cs:38]: We can probably live with this. But the current .NET SDK  can't equate/autoconvert between these two things as in:

```
        ImmutableArray<string> a;
        ImmutableArray<Input<string>> b;

        b = a;

        // MyStack.cs(15,13): error CS0029: Cannot implicitly convert type 'System.Collections.Immutable.ImmutableArray<string>' to 'System.Collections.Immutable.ImmutableArray<Pulumi.Input<string>>' [/Users/anton/tmp/play-cs-pulumi/play-cs-pulumi.csproj]
```

**Review by t0yv0 — APPROVED** [review]
Looking good to me: Go inputs landed, added docs for functions, some unused? output types removed. Added a few comments on some changes that are not obviously bw-compatible.

**Review by stack72 — APPROVED** [review]
Not going to block this but there's an outstanding question on the new c# class numbers

**Comment by komalali** [comment]
@stack72 can you open an issue to track the C# class numbers that you're referring to please so we can prioritize with all the other issues?

**Pull request merged**